apps/api/test/04-report-4.e2e-spec.ts

```diff
@@ -0,0 +1,265 @@
+import { URLS } from './constants';
+import { canActivate } from './override-guard';
+import { getApplicant, getStatus, reportFourNumberOfApplicants } from './report-util';
+import { haPcnArray } from './fixture/reports';
+import { STATUS, LIC_REG_STAGE, BCCNM_LICENSE } from '@ien/common';
+import {
+  addApplicant,
+  addMilestone,
+  setApp,
+  addJob,
+  deleteApplicantStatus,
+} from './report-request-util';
+
+describe('Report 4 - Number of IEN registrants in the licensing stage', () => {
+  // Tracks applicants id's and to later delete status/milestones
+  const applicantIds: string[] = [];
+  const jobIds: string[] = [];
+  let app: INestApplication;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideGuard(AuthGuard)
+      .useValue({ canActivate })
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    setApp(app);
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('Status 200 on report 4 response', async () => {
+    const { status } = await request(app.getHttpServer()).get(URLS.REPORT4);
+    expect(status).toBe(200);
+  });
+
+  it('Add applicants and give each a different status/milestone', async () => {
+    const statusIds: string[] = [];
+
+    const { body: before } = await request(app.getHttpServer()).get(URLS.REPORT4);
+
+    // What the report is expected to look like. Used for comparison at the end of the test
+    const expectedResult = before.map((item: { applicants: string; status: string }) => {
+      const stat = item.status;
+      let result = parseInt(item.applicants) + 1;
+      // Accounts for other two BCCNM Licenses
+      if (stat === 'Granted provisional licensure' || stat === 'Granted full licensure') {
+        result++;
+      }
+      // Withdraw status should not be incremented in report ouput
+      else if (stat === STATUS.WITHDREW_FROM_PROGRAM) {
+        result--;
+      }
+      // NCAS count increases with COMPLETED_CBA and COMPLETED_SLA
+      // NNAS count increases with RECEIVED_NNAS_REPORT and SUBMITTED_DOCUMENTS
+      else if (stat === STATUS.REFERRED_TO_NCAS || stat === STATUS.APPLIED_TO_NNAS) {
+        result += 2;
+      }
+      // Referred to Additional Education makes this report count increase
+      else if (stat === STATUS.COMPLETED_ADDITIONAL_EDUCATION) {
+        result++;
+      }
+      return {
+        ...item,
+        applicants: result.toString(),
+      };
+    });
+
+    // Add applicants and give milestones/status
+    for (const status of LIC_REG_STAGE) {
+      // Job id only needed for WITHDREW_FROM_PROGRAM
+      let jobId = '';
+      const applicant = await addApplicant(getApplicant());
+      applicantIds.push(applicant.id);
+
+      if (status === STATUS.WITHDREW_FROM_PROGRAM) {
+        const { id } = await addJob(applicant.id, {
+          ha_pcn: haPcnArray[0],
```

Comment:
You can refer to the report 6 test using `report-request-util.getHAs`. Also, I wonder if a job is required to add the `withdrew from program` milestone.